util/confutil/config_unix.go

```diff
@@ -0,0 +1,46 @@
+//go:build !windows
+// +build !windows
+
+package confutil
+
+import (
+	"os"
+	"os/user"
+	"strconv"
+)
+
+func init() {
```

Comment:
Put this in `NewConfig` instead of `init`. We can also check that configdir is under the home dir. If ran with `DOCKER_CONFIG=/foobar` then we don't need to convert things I think.

Reply:
> If ran with `DOCKER_CONFIG=/foobar` then we don't need to convert things I think.

Hum right I missed this case. Maybe also check if `DOCKER_CONFIG` is not part of home dir as well.